From 4772991281940efc733fa4d9c45df607de2cc44a Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Sat, 14 May 2016 14:49:48 -0400
Subject: [PATCH 0434/1239] net: thunderx: Copy link parameters from phydev for
 SGMII

This commit makes bgx_poll_for_link() return the correct values
for SGMII mode by copying link status from the phydev structure

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
---
 drivers/net/cavium/thunder_bgx.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/cavium/thunder_bgx.c b/drivers/net/cavium/thunder_bgx.c
index 45ef16ddd8..dd9476e044 100644
--- a/drivers/net/cavium/thunder_bgx.c
+++ b/drivers/net/cavium/thunder_bgx.c
@@ -653,6 +653,10 @@ int bgx_poll_for_link(int node, int bgx_idx, int lmacid)
 			printf("%s: Could not initialize PHY %s\n",
 				lmac->netdev.name, lmac->phydev->dev->name);
 		}
+
+		lmac->link_up = lmac->phydev->link;
+		lmac->last_speed = lmac->phydev->speed;
+		lmac->last_duplex = lmac->phydev->duplex;
 	} else {
 		u64 status1;
 		u64 tx_ctl;
-- 
2.29.0

